src/common/transformations/src/transformations/common_optimizations/ric_fusion.cpp

```diff
@@ -66,7 +66,8 @@ class Attribute {
 
     // Apply callback to materialize RIC inside graph
     void materialize(Input<Node> input, const ov::NodeVector& nodes) const {
-        if (get_axis() >= input.get_partial_shape().size()) {
+        // Despite of m_axis is signed integer this transformartion does not handle negative axes values
+        if (get_axis() < 0 || get_axis() >= static_cast<int64_t>(input.get_partial_shape().size())) {
```

Comment:
if rank is dynamic `input.get_partial_shape().size()` will fail
![image](https://user-images.githubusercontent.com/21181928/204724377-f85b40a2-d74b-4ca0-974b-8408a87b6500.png)

Reply:
Yes, but that's out of the scope of this task - it aims to remove compilation warnings.